cmd/otel-allocator/internal/collector/collector.go

```diff
@@ -124,6 +127,12 @@ func (k *Watcher) runOnCollectors(store cache.Store, fn func(collectors map[stri
 		if pod.Spec.NodeName == "" {
 			continue
 		}
+
+		// pod healthiness check will always be disabled if CollectorNotReadyGracePeriod is set to 0 * time.Second
+		if isPodUnhealthy(pod, k.collectorNotReadyGracePeriod) {
+			continue
+		}
```

Comment:
let's change this to be a method of the watcher i.e. `(k *Watcher) func isPodUnhealthy` ...

Reply:
Sure thing, will fix in next revision. I was thinking I shouldn't add such a `common method` to the class `Watcher`.